### PR TITLE
18358 - Fee Summary component

### PR DESCRIPTION
**Pull request by hfekete**
*Issue:18358 *https://github.com/bcgov/entity/issues/18358 

*Description of changes:*
- added basic fee summary component that can be reused
- added it to layout for the fillings for BTR

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-btr-dev--pr-70-n26nr56e.web.app

File: btr-web/btr-main-app/tests/components/individual-person/SummaryTable.spec.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ describe('AddIndividualPersonSummaryTable tests', () => {
     const year = si.profile.birthDate.getFullYear()
     const month = si.profile.birthDate.getMonth() + 1
     const day = si.profile.birthDate.getDate()
-    expect(details.text()).toContain(`${year}-${month}-${day}`)
+    expect(details.text()).toContain(`${year}-${month.toString().padStart(2, '0')}-${day.toString().padStart(2, '0')}`)
```
Comment (kialj876) [inline btr-web/btr-main-app/tests/components/individual-person/SummaryTable.spec.ts:68]: This is changed a bit in my current PR from Friday so you can remove this

File: btr-web/devops/vaults.env [hunk 2]
```diff
@@ -15,6 +15,8 @@ VUE_APP_REGISTRY_HOME_URL="op://web-url/$APP_ENV/registry/REGISTRY_HOME_URL"
 VUE_APP_AUTH_WEB_URL="op://web-url/$APP_ENV/auth-web/AUTH_WEB_URL"
 
 #vaults API
+#VUE_APP_PAY_API_URL="op://API/$APP_ENV/pay-api/PAY_API_URL"
```
Comment (kialj876) [inline btr-web/devops/vaults.env:18]: uncomment these pls

File: business-transparency-registry.iml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (kialj876) [inline business-transparency-registry.iml:1]: is this for your IDE?

File: btr-web/btr-main-app/app/router.options.ts [hunk 4]
```diff
@@ -40,7 +40,7 @@ export default <RouterConfig> {
           leftButtons: [],
           rightButtons: []
         },
-        layout: 'business'
+        layout: 'filing'
```
Comment (kialj876) [inline btr-web/btr-main-app/app/router.options.ts:43]: we also want this for the route above

File: btr-web/btr-common-components/components/bcros/Widgets/Fee/Index.vue [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+              {{ $t('widgets.feeSummary.total') }}
+            </span>
+            <span class="font-normal text-sm text-gray-400 float-right ml-2">
+              {{ $t('currency.cad') }}
+            </span>
+            <span
+              v-if="fees?.length > 0"
+              class="font-bold text-2xl float-right ml-2 overflow-hidden whitespace-nowrap"
+            >
+              ${{ total }}
```
Comment (kialj876) [inline btr-web/btr-common-components/components/bcros/Widgets/Fee/Index.vue:47]: I think you are missing the dollar signs here and above?

**Review by kialj876 — COMMENTED** [review]
Hey Hrvoje I really like what you've done here. Bunch of small comments, but nice work 👍

File: btr-web/btr-common-components/components/bcros/Widgets/Fee/Index.vue (hunk 5 above)
Comment (hfekete) [reply btr-web/btr-common-components/components/bcros/Widgets/Fee/Index.vue:47]: Hah, I was either missing it above, or had it extra here :P
Updated this now. Can we see examples in the gcbrun ?

File: btr-web/btr-main-app/app/router.options.ts (hunk 4 above)
Comment (hfekete) [reply btr-web/btr-main-app/app/router.options.ts]: Added the widget to the business layout.
Also, you can test adding the fee now, as there is inputs. Changing any form input will trigger adding of the fee.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-btr-dev--pr-70-n26nr56e.web.app

File: btr-web/btr-main-app/app/router.options.ts (hunk 4 above)
Comment (kialj876) [reply btr-web/btr-main-app/app/router.options.ts]: can you add this for when the effective date is added as well?

File: btr-web/btr-layouts/layouts/business.vue [hunk 6]
```diff
@@ -3,8 +3,27 @@
...
     <BcrosBreadcrumb v-if="crumbConstructors.length > 0" :crumb-constructors="crumbConstructors" />
     <BcrosBusinessDetails />
-    <div class="mx-auto px-4 w-full max-w-[1360px]">
-      <slot />
+    <div class="mx-auto p-4 w-full max-w-bcroslg flex justify-center items-start">
+      <div class="max-w-bcrosmd p-4">
+        <slot />
+      </div>
+      <div class="hidden bcroslg:block p-4">
+        <BcrosWidgetsFee :fees="payFeesWidget.fees" data-cy="pay-fees-widget">
```
Comment (kialj876) [inline btr-web/btr-layouts/layouts/business.vue:11]: This should be sticky / move with the screen when you scroll down.

This is an example: https://dev.search.business.bcregistry.gov.bc.ca/open/CP1002539

File: btr-web/devops/vaults.env [hunk 7]
```diff
@@ -15,6 +15,8 @@ VUE_APP_REGISTRY_HOME_URL="op://web-url/$APP_ENV/registry/REGISTRY_HOME_URL"
 VUE_APP_AUTH_WEB_URL="op://web-url/$APP_ENV/auth-web/AUTH_WEB_URL"
 
 #vaults API
+VUE_APP_PAY_API_URL="op://API/$APP_ENV/pay-api/PAY_API_URL"
+VUE_APP_PAY_API_VERSION="op://API/$APP_ENV/pay-api/PAY_API_URL  "
```
Comment (kialj876) [inline btr-web/devops/vaults.env:19]: got some extra spaces in the quote here

**Review by kialj876 — COMMENTED** [review]
last couple things:

File: btr-web/btr-common-components/components/bcros/inputs/DateSelect.vue [hunk 8]
```diff
@@ -45,9 +46,8 @@ const updateDate = (val: Date | null) => {
   selectedDate.value = val
 }
 
-const selectedDateDisplay: ComputedRef<string> = computed(() => selectedDate.value?.toLocaleDateString() || '')
+const selectedDateDisplay: ComputedRef<string> = computed(
+  () => selectedDate.value ? moment(selectedDate.value).format('YYYY-MM-DD') : ''
```
Comment (kialj876) [inline btr-web/btr-common-components/components/bcros/inputs/DateSelect.vue:50]: this is actually going against the design :P , can you change it back?

File: btr-web/btr-main-app/cypress/e2e/pages/ownerChange.cy.ts [hunk 9]
```diff
@@ -26,8 +26,13 @@ describe('pages -> Beneficial Owner Change', () => {
...
     cy.get('[data-cy=date-picker]').get('.bcros-date-picker__calendar__day').eq(0).trigger('click')
-    const today = moment(new Date()).format('MM/D/YYYY')
-    cy.get('[data-cy=date-select]').should('have.value', today)
+
+    const today = new Date()
+    const year = today.getFullYear()
+    const monty = today.getMonth()
+    const day = 1
+    const expectedDate = moment(new Date(year, monty, day)).format('YYYY-MM-DD')
+    cy.get('[data-cy=date-select]').should('have.value', expectedDate)
```
Comment (kialj876) [inline btr-web/btr-main-app/cypress/e2e/pages/ownerChange.cy.ts:35]: this test is already fixed if you revert and rebase the most recent changes fyi

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-btr-dev--pr-70-n26nr56e.web.app

File: btr-web/btr-common-components/components/bcros/inputs/DateSelect.vue (hunk 8 above)
Comment (hfekete) [reply btr-web/btr-common-components/components/bcros/inputs/DateSelect.vue]: updated this due to changes in design.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-btr-dev--pr-70-n26nr56e.web.app

**Review by kialj876 — APPROVED** [review]
Nice work 👍

**Pull request merged**